Structural/Decorator/JsonRenderer.php

```diff
@@ -8,4 +8,16 @@ public function renderData(): string
     {
         return json_encode($this->wrapped->renderData());
     }
+
+    /**
+     * Renders data as JSON based on options passed.
+     * In terms of decorator patters, it represents additional extended behaviour.
+     *
+     * @param int $options Options to be used to encode data into JSON.
+     * @return string Rendered output
+     */
+    public function renderDataWithOptions($options = 0): string
```

Comment:
@adayth the idea here is to show that we can add some additional method (behaviour) to the class